### PR TITLE
Distinguish witnesses and divulgences in scenario table view

### PR DESCRIPTION
**Pull request by hurryabit**
Currently, we display a "D" (for "Disclosure/Divulgence") in the table
view of the scenario view when a party can see a contract because they
have learnt about it via disclosure.

This change further refines this visibily indicators for disclosure:
We display a "W" for "Witness" (i.e. the party has witnessed the
creation of the contract) and "D" for "Divulgence".

Help for the reviewer: In the scenario service server, witnessing is
called explicit disclosure whereas divulgence is called implicit
disclosure.

CHANGELOG_BEGIN
* [DAML Studio]
  When displaying scenario results in table view in DAML Studio, we now
  indicate _why_ a party knows about the existence of a contract:
  - `S` means the party is a signatory.
  - `O` means the party is an observer.
  - `W` means the party has witnessed the creation of the contract.
  - `D` means the party has learned about the contract via divulgence.
CHANGELOG_END

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

---
This change is [Reviewable](https://reviewable.io/reviews/digital-asset/daml/6571)

### CODE REVIEW

**Comment by hurryabit** [comment]
Screenshot 2020-07-02 at 08 05 31

File: daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/scenario/ScenarioLedger.scala [hunk 1]
```diff
@@ -83,7 +83,7 @@ object ScenarioLedger {
       globalImplicitDisclosure: Relation[ContractId, Party],
       failedAuthorizations: FailedAuthorizations,
   ) {
-    def disclosures(coidToEventId: ContractId => EventId): Relation[EventId, Party] =
+    def disclosures(coidToEventId: ContractId => EventId): Map[EventId, Set[Party]] =
```
Comment (cocreature) [inline daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/scenario/ScenarioLedger.scala:86]: While I’m not really a fan of synonyms it also seems a bit confusing to work with `Relation.…` in the body of this method but then not use the type synonym.

File: daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/scenario/ScenarioLedger.scala [hunk 2]
```diff
@@ -521,9 +526,18 @@ object ScenarioLedger {
...
+      // NOTE(MH): Since `addDisclosures` is biased towards existing
+      // disclosures, we need to add the "stronger" explicit ones first.
+      val cacheWithExplicitDisclosures =
+        richTr.explicitDisclosure.foldLeft(cacheAfterProcess) {
+          case (cacheP, (nodeId, witnesses)) =>
+            cacheP.updateLedgerNodeInfo(EventId(richTr.transactionId, nodeId))(
+              _.addDisclosures(witnesses.map(_ -> Disclosure(since = trId, explicit = true)).toMap))
+        }
+      richTr.globalImplicitDisclosure.foldLeft(cacheWithExplicitDisclosures) {
+        case (cacheP, (coid, witnesses)) =>
```
Comment (cocreature) [inline daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/scenario/ScenarioLedger.scala:538]: Shouldn’t `witnesses` be called `divulgees` here?

File: daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/scenario/ScenarioLedger.scala (hunk 1 above)
Comment (hurryabit) [reply daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/scenario/ScenarioLedger.scala:86]: Good catch. That's a relic from a failed attempt to change some things. I'll bring it back to its original state. I should have self-reviewed my PR better. 😞

File: daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/scenario/ScenarioLedger.scala (hunk 2 above)
Comment (hurryabit) [reply daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/scenario/ScenarioLedger.scala:538]: That's a good idea. Thanks. I blindly copied the old code...

**Pull request merged**